docs/getting-started/provider-gcp.md

```diff
@@ -0,0 +1,227 @@
+
+You can view the new CRDs with `kubectl get crds`. Every CRD maps to a unique GCP service Crossplane can provision and manage.
+
+{{< hint type="tip" >}}
+All the supported CRDs are also available in the [Upbound Marketplace](https://marketplace.upbound.io/providers/upbound/provider-gcp/latest/crds).
+{{< /hint >}}
+
+### Create a Kubernetes secret for GCP
+The provider requires credentials to create and manage GCP resources. Providers use a Kubernetes _Secret_ to connect the credentials to the provider.
+
+First generate a Kubernetes _Secret_ from a Google Cloud service account JSON file and then configure the Provider to use it.
+
+{{< hint type="note" >}}
+Other authentication methods exist and are beyond the scope of this guide. The [Provider documentation](https://marketplace.upbound.io/providers/upbound/provider-gcp/latest/docs/configuration) contains information on alternative authentication methods. 
+{{< /hint >}}
+
+#### Generate a GCP service account JSON file
+For basic user authentication, use a Google Cloud service account JSON file. 
+
+{{< hint type="tip" >}}
+The [GCP documentation](https://cloud.google.com/iam/docs/creating-managing-service-account-keys) provides information on how to generate a service account JSON file.
+{{< /hint >}}
+
+Save this JSON file as `gcp-credentials.json`
+
+{{< hint type="note" >}}
+The [Configuration](https://marketplace.upbound.io/providers/upbound/provider-gcp/latest/docs/configuration) section of the Provider documentation describes other authentication methods.
+{{< /hint >}}
+
+#### Create a Kubernetes secret with the GCP credentials
+<!-- vale gitlab.Substitutions = NO -->
+<!-- ignore .json file name -->
+A Kubernetes generic secret has a name and contents. Use {{< hover label="kube-create-secret" line="1">}}kubectl create secret{{< /hover >}} to generate the secret object named {{< hover label="kube-create-secret" line="2">}}gcp-secret{{< /hover >}} in the {{< hover label="kube-create-secret" line="3">}}crossplane-system{{</ hover >}} namespace.  
+Use the {{< hover label="kube-create-secret" line="4">}}--from-file={{</hover>}} argument to set the value to the contents of the  {{< hover label="kube-create-secret" line="4">}}gcp-credentials.json{{< /hover >}} file.
+<!-- vale gitlab.Substitutions = YES -->
+
+```shell {label="kube-create-secret",copy-lines="all"}
+kubectl create secret \
+generic gcp-secret \
+-n crossplane-system \
+--from-file=creds=./gcp-credentials.json
+```
+
+View the secret with `kubectl describe secret`
+
+{{< hint type="note" >}}
+The size may be larger if there are extra blank spaces in your text file.
+{{< /hint >}}
+
+```shell
+kubectl describe secret gcp-secret -n crossplane-system
+Name:         gcp-secret
+Namespace:    crossplane-system
+Labels:       <none>
+Annotations:  <none>
+
+Type:  Opaque
+
+Data
+====
+creds:  2330 bytes
```

Comment:
This isn't a valid snippet of Bourne shell.

Reply:
Same as before, this is a hint to the syntax highlighter, not an indication to the actual shell usage.

Reply:
For Kubernetes docs, we [separate commands from output](https://kubernetes.io/docs/contribute/style/style-guide/#separate-commands-from-output).

Reply:
I'm making a different editorial choice. It's easier to see the command and output in a single place. The main motivator for separating the command from output would be to simplify copy/paste. That will be address in the Crossplane docs in the near future with a copy button that allows the author to define which lines in the output the button copies.

Reply:
Seems reasonable. It'll only really be a problem if you get a large team of contributors and need to codify things for a style guide.

Reply:
(feel free to mark this resolved)

Reply:
Agreed! With some of the upcoming style changes I am planning an large expansion to the existing [Contributor's Guide](http://localhost:1313/docs/master/contributing/docs.html). You can see some of that expansion today in the [Upbound Docs Contributor's Guide](https://docs.upbound.io/contributing).